### PR TITLE
feat(guppy): add test plan for guppy nested query

### PR DESCRIPTION
**Pull request by qingyashu**
See this PR: https://github.com/uc-cdis/guppy/pull/52  (WIP)

### CODE REVIEW

File: suites/guppy/nestedQueryTest.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+ * 1. Data in the commons are ETLed
+ * 2. Data has a nested structure. Take qa-mickey as example:
+ * "patient" doc has nested prop "ActionableMutations"
+ * 3. Guppy is up, take qa-mickey as example:
+          guppy version entry: https://github.com/uc-cdis/gitops-qa/blob/master/qa-mickey.planx-pla.net/manifest.json#L23
+          config: https://github.com/uc-cdis/gitops-qa/blob/master/qa-mickey.planx-pla.net/manifest.json#L96-L104
+          related portal config: https://github.com/uc-cdis/gitops-qa/blob/master/qa-mickey.planx-pla.net/portal/gitops.json#L223-L228
+ * 4. User got access to the website and the data, and could visit `/query` page
+ */
+Feature('Support query for nested data - https://github.com/uc-cdis/guppy/pull/52');
```
Comment (themarcelor) [inline suites/guppy/nestedQueryTest.js]: We usually reference the JIRA url here, right?
e.g.,
https://github.com/uc-cdis/gen3-qa/pull/175/files#diff-00a19f5fc7c95ed6f4a05ff5721adc5cR1

File: suites/guppy/nestedQueryTest.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+const chai = require('chai');
+const { interactive, ifInteractive } = require('../../utils/interactive.js');
+
+const { expect } = chai;
+
+Scenario('Check guppy is up and query page is ready', ifInteractive(
```
Comment (frickjack) [inline suites/guppy/nestedQueryTest.js]: Please add a `@manual` annotation - that way the test won't run (and just always succeed) in Jenkins :+1:

**Review by frickjack — COMMENTED** [review]
Very cool!

File: suites/guppy/nestedQueryTest.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    const result = await interactive(`
+        1. Goto target commons (e.g., https://qa-mickey.planx-pla.net/).
+        2. Log in, and then click "Query" in nav bar.
+        3. You should be able to see GraphiQL editor, the result panel in the right side should not have any error messages. 
+    `);
+    expect(result.didPass, result.details).to.be.true;
+  },
+));
+
+Scenario('could query nested data', ifInteractive(
```
Comment (themarcelor) [inline suites/guppy/nestedQueryTest.js]: the executable test scenarios must contain the CodeceptJS tag " @manual "
So they won't run as part of the fully automated PR checks (integration tests).|

More details in this blog post:
https://ctds-planx.atlassian.net/wiki/spaces/~696307876/blog/2019/10/24/143949839/How+we+test+stuff#Including-and-Excluding-tests

File: suites/guppy/nestedQueryTest.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+                    }
+                  ]
+                }
+              }
+    `);
+    expect(result.didPass, result.details).to.be.true;
+  },
+));
+
+Scenario('could query data with a filter that contains nested data', ifInteractive(
```
Comment (themarcelor) [inline suites/guppy/nestedQueryTest.js]: missing `@manual` tag

File: suites/guppy/nestedQueryTest.js [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                    }
+                  ]
+                }
+              }
+    `);
+    expect(result.didPass, result.details).to.be.true;
+  },
+));
+
+Scenario('get query with a complicated filters that contains two or more nested props', ifInteractive(
```
Comment (themarcelor) [inline suites/guppy/nestedQueryTest.js]: missing `@manual` tag

File: suites/guppy/nestedQueryTest.js [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                    }
+                  ]
+                }
+              }
+    `);
+    expect(result.didPass, result.details).to.be.true;
+  },
+));
+
+Scenario('get data with a complicated filters that combines nested props and normal props', ifInteractive(
```
Comment (themarcelor) [inline suites/guppy/nestedQueryTest.js]: missing `@manual` tag

File: suites/guppy/nestedQueryTest.js [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+ * Prerequisites:
+ * 1. Data in the commons are ETLed
+ * 2. Data has a nested structure. Take qa-mickey as example:
+ * "patient" doc has nested prop "ActionableMutations"
+ * 3. Guppy is up, take qa-mickey as example:
+          guppy version entry: https://github.com/uc-cdis/gitops-qa/blob/master/qa-mickey.planx-pla.net/manifest.json#L23
+          config: https://github.com/uc-cdis/gitops-qa/blob/master/qa-mickey.planx-pla.net/manifest.json#L96-L104
+          related portal config: https://github.com/uc-cdis/gitops-qa/blob/master/qa-mickey.planx-pla.net/portal/gitops.json#L223-L228
+ * 4. User got access to the website and the data, and could visit `/query` page
+ */
```
Comment (atharvar28) [inline suites/guppy/nestedQueryTest.js]: Can you move this block inside the test? 
The whole idea of pre-req is to help the user/tester to understand what initial setup needs to be done, we want the testers to see this!

**Review by atharvar28 — APPROVED** [review]
lgtm 👍

**Review by themarcelor — APPROVED** [review]
lgtm

**Pull request merged**